### ISSUE TITLE
Build fail on i386

### ISSUE THREAD

**Issue by pvizeli** [Legrandin/pycryptodome#123]
On alpine linux 3.7 i386 it fails with follow error (With 3.4.5 it work correct):

```
/usr/lib/gcc/i586-alpine-linux-musl/6.4.0/include/emmintrin.h:637:1: error:
inlining failed in call to always_inline '_mm_set1_epi32': target specific optio
n mismatch
     _mm_set1_epi32 (int __A)
     ^~~~~~~~~~~~~~
    src/multiply_32.c:64:8: note: called from here
         r0 = _mm_set1_epi32(b);             // { b, b, b, b }
         ~~~^~~~~~~~~~~~~~~~~~~
    In file included from /usr/lib/gcc/i586-alpine-linux-musl/6.4.0/include/xmmi
ntrin.h:1249:0,
                     from /usr/lib/gcc/i586-alpine-linux-musl/6.4.0/include/x86i
ntrin.h:33,
                     from src/multiply_32.c:39:
    /usr/lib/gcc/i586-alpine-linux-musl/6.4.0/include/emmintrin.h:1539:1: error:
 inlining failed in call to always_inline '_mm_castps_si128': target specific op
tion mismatch
     _mm_castps_si128(__m128 __A)
```

**Comment by Legrandin** [comment]
Can you  post also the gcc command line that breaks with this error (compiling `multiply_32.c`)?

I have Alpine Linux 3.7 i386 (gcc 6.4.0, python 2.7.14) and it compiles fine.

**Comment by pvizeli** [comment]
`pip3 install pycryptodome`

**Comment by pvizeli** [comment]
First is with command above and second with old command.

![image](https://user-images.githubusercontent.com/15338540/35617548-c8c3360c-0678-11e8-9266-53a9218a46d8.png)

**Comment by Legrandin** [comment]
Thanks. My guess is that:

    python -c "import platform; print platform.machine()"

shows something else than `i686`. If that's the case, I have a patch ready.

**Comment by pvizeli** [comment]
![image](https://user-images.githubusercontent.com/15338540/35671175-db691dd6-073a-11e8-9055-0fe3a74d0f3f.png)

Hmm, we have no real i386 (i686) hardware, we run it as crossbuild on a amd64 machine...

**Comment by Legrandin** [comment]
This should be fixed in v3.4.9. Closing.
